fem/libceed/full-assembly.cpp

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "full-assembly.hpp"
+
+#ifdef MFEM_USE_CEED
+#include "util.h"
+
+int CeedHackReallocArray(size_t n, size_t unit, void *p)
+{
+   *(void **)p = realloc(*(void **)p, n*unit);
+   if (n && unit && !*(void **)p)
+      return CeedError(NULL, 1, "realloc failed to allocate %zd members of size "
+                       "%zd\n", n, unit);
+   return 0;
+}
+
+#define CeedHackRealloc(n, p) CeedHackReallocArray((n), sizeof(**(p)), p)
+
+int CeedHackFree(void *p)
+{
+   free(*(void **)p);
+   *(void **)p = NULL;
+   return 0;
+}
```

Comment:
Is the name already indicating that a reviewer should ask about this? :)

Reply:
Yes, and we've already had some discussions here. The hope is to eventually contribute the full assembly code to the Ceed library, in which case it will use `CeedRealloc` and `CeedFree`. Those are inaccessible in the MFEM context. We discussed contributing to Ceed first, then doing this PR, but decided it made sense to get everything up and running in MFEM first.